driver/csiplugin/connectors/rest_v2.go

```diff
@@ -702,6 +796,55 @@ func (s *SpectrumRestV2) ListFileset(ctx context.Context, filesystemName string,
 	return getFilesetResponse.Filesets[0], nil
 }
 
+func (s *SpectrumRestV2) CheckFilesetWithAFMTarget(ctx context.Context, filesystemName string, afmTarget string) (string, error) {
+	loggerID := utils.GetLoggerId(ctx)
+	klog.V(4).Infof("[%s] rest_v2 CheckFilesetWithAFMTarget. filesystem: %s, afmTarget: %s", loggerID, filesystemName, afmTarget)
+
+	url := fmt.Sprintf("scalemgmt/v2/filesystems/%s/filesets?fields=afm&filter=config.isInodeSpaceOwner=true", filesystemName)
+	klog.V(6).Infof("[%s] getFilesetURL [%v] ", loggerID, url)
+	getFilesetResponse := GetFilesetResponse_v2{}
+
+	err := s.doHTTP(ctx, url, "GET", &getFilesetResponse, nil)
+	if err != nil {
+		klog.Errorf("[%s] Error in list fileset request with the field AFM: %v", loggerID, err)
+		return "", err
+	}
+
+	emptyAFM := AFM{}
+	// TODO: Optimize this when GUI has a filter for AFM
+	// Check if cache fileset with the same bucket exists in the first response.
+	for _, fileset := range getFilesetResponse.Filesets {
```

Comment:
Yes, it's required, first loop is for checking the first response from GUI, if we get the required fileset, we will return from there itself, otherwise we need to make another call to get next pages and check the same.